### PR TITLE
[JAV-575] Exception to response converter

### PR DESCRIPTION
**Pull request by wujimin**

### CODE REVIEW

File: swagger/swagger-invocation/invocation-core/src/main/java/io/servicecomb/swagger/invocation/exception/ExceptionToResponseConverters.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  @SuppressWarnings("unchecked")
+  public ExceptionToResponseConverters() {
+    SPIServiceUtils.getAllService(ExceptionToResponseConverter.class).forEach(converter -> {
+      exceptionToResponseConverters.put(converter.getExceptionClass(), converter);
+    });
+  }
+
+  public Response convertExceptionToResponse(SwaggerInvocation swaggerInvocation, Throwable e) {
+    ExceptionToResponseConverter<Throwable> converter = exceptionToResponseConverters.get(e.getClass());
+    if (converter == null) {
```
Comment (WillemJiang) [inline swagger/swagger-invocation/invocation-core/src/main/java/io/servicecomb/swagger/invocation/exception/ExceptionToResponseConverters.java]: Do we have default converter to handler the exception?

File: swagger/swagger-invocation/invocation-core/src/test/java/io/servicecomb/swagger/engine/TestSwaggerProducerOperation.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+public class TestSwaggerProducerOperation {
+  SwaggerProducerOperation swaggerProducerOperation = new SwaggerProducerOperation();
+
+  @Test
+  public void processException_convert(@Mocked SwaggerInvocation invocation) {
+    Error error = new Error("abc");
+
+    Response response = swaggerProducerOperation.processException(invocation, error);
+    Assert.assertSame(Status.OK, response.getStatus());
```
Comment (WillemJiang) [inline swagger/swagger-invocation/invocation-core/src/test/java/io/servicecomb/swagger/engine/TestSwaggerProducerOperation.java]: If there is an exception, do we need to set the status to be OK?

Comment (wujimin) [reply swagger/swagger-invocation/invocation-core/src/test/java/io/servicecomb/swagger/engine/TestSwaggerProducerOperation.java]: this is just a test demo.
converter will not limit the convert logic, some customers(eg: music) will convert a exception to normal response, and error information be setted to body model.

File: swagger/swagger-invocation/invocation-core/src/main/java/io/servicecomb/swagger/invocation/exception/ExceptionToResponseConverters.java (hunk 1 above)
Comment (wujimin) [reply swagger/swagger-invocation/invocation-core/src/main/java/io/servicecomb/swagger/invocation/exception/ExceptionToResponseConverters.java]: default logic is in swaggerProducerOperation
that's not so good, i will extract it to be a default converter

Comment (wujimin) [reply swagger/swagger-invocation/invocation-core/src/main/java/io/servicecomb/swagger/invocation/exception/ExceptionToResponseConverters.java]: done

**Pull request merged**